src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/ListingGenerationController.java

```diff
@@ -322,20 +323,19 @@ public ResponseEntity<ListingCallbackResponse> generateHearingDocument(
         var listingData = listingRequest.getCaseDetails().getCaseData();
         var caseTypeId = listingRequest.getCaseDetails().getCaseTypeId();
         List<String> errorsList = new ArrayList<>();
-        boolean invalidCharsExist = listingService.checkInvalidCharsForAllParties(
-                listingRequest.getCaseDetails(), userToken, errorsList);
-        boolean listingsExist = hasListings(listingData);
-        if (!invalidCharsExist && listingsExist) {
+        boolean invalidCharsExist = InvalidCharacterCheck.invalidCharactersExistAllListingTypes(
+                listingRequest.getCaseDetails(), errorsList);
+        if (!invalidCharsExist && !hasListings(listingData)) {
+                errorsList.add("No cases with hearings have been found for your search criteria");
```

Comment:
With this - if a case has invalid character but hearings exist, you would see the message no cases with hearings have been found despite it being a character issue. Just noticed it from another glance